Doc/tools/extensions/pyspecific.py

```diff
@@ -393,7 +394,23 @@ def run(self):
         return PyMethod.run(self)
 
 
-# Support for documenting version of removal in deprecations
+# Support for documenting version of changes, additions, deprecations
+
+def expand_version_arg(argument, env):
+    """Expand "next" to the current version"""
+    if argument == 'next':
+        return sphinx_gettext('{} (unreleased)').format(env.config.release)
+    return argument
+
+
+class PyVersionChange(sphinx.directives.other.VersionChange):
+    def run(self):
+        env = self.state.document.settings.env
+        self.arguments = (
+            expand_version_arg(self.arguments[0], env),
+            *self.arguments[1:])
```

Comment:
The ``.config`` shortcut [exists from Sphinx 1.8](https://www.sphinx-doc.org/en/master/extdev/utils.html#sphinx.util.docutils.SphinxDirective.config).

```suggestion
        # Replace the 'next' special token with the current development version
        self.arguments[0] = expand_version_arg(self.arguments[0], self.config.release)
```